### PR TITLE
NodeJS node_options variable and Kibana plugin optimization

### PR DESCRIPTION
**Pull request by Zenidd**
Hi team,

- This PR adds a new variable (`node_options`) that allows a space-separated list of command-line options to be added so they are taken into account by the Kibana launch script.

- On the other hand, it also adds a new task for Kibana plugin optimization into the `ansible-kibana` set of tasks.

Greetings, JP Sáez

### CODE REVIEW

**Comment by andymotta** [comment]
I had also enabled options for `--max-old-space-size` before seeing this feature branch.  This is the behaviour we were seeing with out of the box defaults:

```bash
Mar 10 18:26:25 wazuh-server kibana[2609]: {"type":"log","@timestamp":"2020-03-10T18:26:25Z","tags":["info","optimize"],"pid":2609,"message":"Optimizing and caching bundles for core, graph, monitoring, space_selector, login, overwritten_session, logout, logged_out, ml, dashboardViewer, apm, maps, canvas, infra, siem, uptime, lens, wazuh, kibana, stateSessionStorageRedirect, status_page and timelion. This may take a few minutes"}
Mar 10 18:30:58 wazuh-server kibana[2609]: <--- Last few GCs --->
Mar 10 18:30:58 wazuh-server kibana[2609]: [2609:0x3b20f40]   288840 ms: Mark-sweep 1300.4 (1397.9) -> 1300.4 (1397.9) MB, 1743.9 / 0.0 ms  (average mu = 0.267, current mu = 0.000) last resort GC in old space requested
Mar 10 18:30:58 wazuh-server kibana[2609]: [2609:0x3b20f40]   290619 ms: Mark-sweep 1300.4 (1397.9) -> 1300.4 (1397.9) MB, 1779.1 / 0.0 ms  (average mu = 0.150, current mu = 0.000) last resort GC in old space requested
Mar 10 18:30:58 wazuh-server kibana[2609]: <--- JS stacktrace --->
Mar 10 18:30:58 wazuh-server kibana[2609]: ==== JS stack trace =========================================
Mar 10 18:30:58 wazuh-server kibana[2609]:     0: ExitFrame [pc: 0x3d5550e5be1d]
Mar 10 18:30:58 wazuh-server kibana[2609]: Security context: 0x0d8c7a21e6e9 <JSObject>
Mar 10 18:30:58 wazuh-server kibana[2609]:     1: add [0xd8c7a211831](this=0x1fed594415e9 <Set map = 0x3222ba4841f9>,0x2463011e4141 <JSArray[0]>)
Mar 10 18:30:58 wazuh-server kibana[2609]:     2: prepend_comments [0x1ee055e0f549] [/usr/share/kibana/node_modules/terser/dist/bundle.min.js:~1] [pc=0x3d5551ee6841](this=0x029e5678a7e9 <Object map = 0x12fd8331d321>,t=0x0419d287df99 <AST_Dot map = 0x1e39db865e61>)
Mar 10 18:30:58 wazuh-server kibana[2609]:     3: /* anonymous */(aka /* anony...
Mar 10 18:30:58 wazuh-server kibana[2609]: FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
Mar 10 18:30:58 wazuh-server kibana[2609]:  1: 0x8fa090 node::Abort() [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  2: 0x8fa0dc  [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  3: 0xb0052e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  4: 0xb00764 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  5: 0xef4c72  [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  6: 0xf0448f v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  7: 0xecd4e6 v8::internal::Factory::AllocateRawArray(int, v8::internal::PretenureFlag) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  8: 0xecdd6a v8::internal::Factory::NewFixedArrayWithFiller(v8::internal::Heap::RootListIndex, int, v8::internal::Object*, v8::internal::PretenureFlag) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]:  9: 0xecde10 v8::internal::Handle<v8::internal::FixedArray> v8::internal::Factory::NewFixedArrayWithMap<v8::internal::FixedArray>(v8::internal::Heap::RootListIndex, int, v8::internal::PretenureFlag) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]: 10: 0x1059df7 v8::internal::OrderedHashTable<v8::internal::OrderedHashSet, 1>::Allocate(v8::internal::Isolate*, int, v8::internal::PretenureFlag) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]: 11: 0x105d0bb v8::internal::OrderedHashTable<v8::internal::OrderedHashSet, 1>::Rehash(v8::internal::Handle<v8::internal::OrderedHashSet>, int) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]: 12: 0x105d689 v8::internal::OrderedHashTable<v8::internal::OrderedHashSet, 1>::EnsureGrowable(v8::internal::Handle<v8::internal::OrderedHashSet>) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]: 13: 0x1141dce v8::internal::Runtime_SetGrow(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/share/kibana/bin/../node/bin/node]
Mar 10 18:30:58 wazuh-server kibana[2609]: 14: 0x3d5550e5be1d
```

File: roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml [hunk 1]
```diff
@@ -63,9 +63,7 @@
       chdir: "/tmp/app/build"
 
   - name: Install Wazuh Plugin (can take a while)
-    shell: "/usr/share/kibana/bin/kibana-plugin install file:///tmp/app/build/{{ wazuhapp_package_name.stdout }}"
-    environment:
-      NODE_OPTIONS: "--max-old-space-size=3072"
+    shell: NODE_OPTIONS=" {{ node_options }} " /usr/share/kibana/bin/kibana-plugin install file:///tmp/app/build/{{ wazuhapp_package_name.stdout }}
```
Comment (xr09) [inline roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml]: No need for the extra spaces, you can use: `NODE_OPTIONS="{{ node_options }}"`

File: roles/elastic-stack/ansible-kibana/tasks/main.yml [hunk 2]
```diff
@@ -119,9 +131,9 @@
 
 
 - name: Install Wazuh Plugin (can take a while)
-  shell: "/usr/share/kibana/bin/kibana-plugin install {{ wazuh_app_url }}-{{ wazuh_version }}_{{ elastic_stack_version }}.zip"
-  environment:
-    NODE_OPTIONS: "--max-old-space-size=3072"
+  shell: >-
+    NODE_OPTIONS=" {{ node_options }} " /usr/share/kibana/bin/kibana-plugin install
```
Comment (xr09) [inline roles/elastic-stack/ansible-kibana/tasks/main.yml]: Same here

File: roles/elastic-stack/ansible-kibana/tasks/main.yml [hunk 3]
```diff
@@ -135,6 +147,16 @@
   when:
     - not build_from_sources
 
+- name: Kibana optimization (can take a while)
+  shell: NODE_OPTIONS=" {{ node_options }} " /usr/share/kibana/bin/kibana --optimize
```
Comment (xr09) [inline roles/elastic-stack/ansible-kibana/tasks/main.yml]: Here too

Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/main.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

File: roles/elastic-stack/ansible-kibana/tasks/main.yml (hunk 2 above)
Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/main.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

File: roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml (hunk 1 above)
Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

**Comment by Zenidd** [comment]
Hello @andymotta,

Several users have been reporting to us in our [Slack community](https://wazuh.com/community/join-us-on-slack/) channel that they encountered this problem during kibana optimization. 

Thank you for let us know that this is not something happening just in our lab environments. Please feel free to contribute other solutions or workarounds that worked in your environment in any other context.


Greetings, 
JP Sáez

File: roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml (hunk 1 above)
Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/build_wazuh_plugin.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

File: roles/elastic-stack/ansible-kibana/tasks/main.yml (hunk 2 above)
Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/main.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

File: roles/elastic-stack/ansible-kibana/tasks/main.yml (hunk 3 above)
Comment (Zenidd) [reply roles/elastic-stack/ansible-kibana/tasks/main.yml]: The extra spaces have been removed on https://github.com/wazuh/wazuh-ansible/pull/385/commits/84b5510e3010f88da3863a53d416c789786fbded

**Pull request merged**